actionpack/test/controller/http_basic_authentication_test.rb

```diff
@@ -170,6 +170,12 @@ def test_encode_credentials_has_no_newline
     assert_response :unauthorized
   end
 
+  test "authentication request with credentials without colon" do
+    @request.env["HTTP_AUTHORIZATION"] = "Basic #{::Base64.encode64("David Goliath")}"
+    get :search
+    assert_response :unauthorized
+  end
+
```

Comment:
As the credentials "David Goliath" are invalid as well, maybe it's better to move the test to a specific `has_basic_credentials?` test like we do for `encode_credentials`:
https://github.com/rails/rails/blob/5462fbd5de1900c1b1ce1c9dc11c1a2d8cdcd809/actionpack/test/controller/http_basic_authentication_test.rb#L107-L113

Something like:
```suggestion
  test "authentication request with credentials without colon" do
    @request.env["HTTP_AUTHORIZATION"] = "Basic #{::Base64.encode64("David Goliath")}"
    refute ActionController::HttpAuthentication::Basic.has_basic_credentials(@request)
  end
```

Reply:
Ok, I changed the test to specific for `has_basic_credentials?`.

Reply:
Thanks! 😄 I have one last change request.
Could you move it under the `test_encode_credentials_has_no_newline` test?
This groups the tests that don't do `get` requests together.

Reply:
Yes, of course! :slightly_smiling_face: I moved it.

Reply:
Thanks 😄